src/dotnet/UI.Blazor/Components/Modal/BlazoredModal/BlazoredModal.razor

```diff
@@ -0,0 +1,165 @@
+﻿@using System.Collections.ObjectModel
+@using Microsoft.AspNetCore.Components.Routing
+@namespace Blazored.Modal
+
+@inject NavigationManager NavigationManager
+@inject IJSRuntime JsRuntime
+@implements IAsyncDisposable
+
+<CascadingValue Value="this">
+    <CascadingValue Value="_globalModalOptions">
+        @foreach (var modal in _modals)
+        {
+            @modal.ModalInstance
+        }
+    </CascadingValue>
+</CascadingValue>
+
+@code {
+    [CascadingParameter] private IModalService CascadedModalService { get; set; } = default!;
+
+    [Parameter] public bool? HideHeader { get; set; }
+    [Parameter] public bool? HideCloseButton { get; set; }
+    [Parameter] public bool? DisableBackgroundCancel { get; set; }
+    [Parameter] public string? OverlayCustomClass { get; set; }
+    [Parameter] public ModalPosition? Position { get; set; }
+    [Parameter] public string? PositionCustomClass { get; set; }
+    [Parameter] public ModalSize? Size { get; set; }
+    [Parameter] public string? SizeCustomClass { get; set; }
+    [Parameter] public string? Class { get; set; }
+    [Parameter] public ModalAnimationType? AnimationType { get; set; }
+    [Parameter] public bool? UseCustomLayout { get; set; }
+    [Parameter] public bool? ActivateFocusTrap { get; set; }
+
+    private readonly Collection<ModalReference> _modals = new();
+    private readonly ModalOptions _globalModalOptions = new();
+    private IJSObjectReference? _styleFunctions;
+    private bool _haveActiveModals;
+
+    internal event Action? OnModalClosed;
+
+    protected override void OnInitialized()
+    {
+        if (CascadedModalService == null)
+        {
+            throw new InvalidOperationException($"{GetType()} requires a cascading parameter of type {nameof(IModalService)}.");
+        }
+
+        ((ModalService) CascadedModalService).OnModalInstanceAdded += Update;
+        ((ModalService) CascadedModalService).OnModalCloseRequested += CloseInstance;
+        NavigationManager.LocationChanged += CancelModals;
+
+        _globalModalOptions.Class = Class;
+        _globalModalOptions.DisableBackgroundCancel = DisableBackgroundCancel;
+        _globalModalOptions.HideCloseButton = HideCloseButton;
+        _globalModalOptions.HideHeader = HideHeader;
+        _globalModalOptions.Position = Position;
+        _globalModalOptions.PositionCustomClass = PositionCustomClass;
+        _globalModalOptions.Size = Size;
+        _globalModalOptions.SizeCustomClass = SizeCustomClass;
+        _globalModalOptions.AnimationType = AnimationType;
+        _globalModalOptions.OverlayCustomClass = OverlayCustomClass;
+
+        _globalModalOptions.UseCustomLayout = UseCustomLayout;
+        _globalModalOptions.ActivateFocusTrap = ActivateFocusTrap;
+    }
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (firstRender)
+        {
```

Comment:
!

Reply:
Lots of such things, actually... Won't comment on these further.

Reply:
fixed